tests/stack_test.go

```diff
@@ -456,6 +457,25 @@ func TestLocalStateLocking(t *testing.T) {
 			}
 		}
 	}
+
+	// Run 10 concurrent previews
```

Comment:
Added test coverage here - but this test is skipped currently due to #7269.  I verified locally that this test correctly fails before the change and succeeds after the change when un-skipped.